toolchain/sem_ir/typed_nodes.h

```diff
@@ -129,6 +129,15 @@ struct BoolLiteral {
   BoolValue value;
 };
 
+struct BoundMethod {
+  static constexpr auto Kind = NodeKind::BoundMethod.Define("bound_method");
+
+  Parse::Node parse_node;
+  TypeId type_id;
+  NodeId object_id;
```

Comment:
Minor question, I assume we'll use this for interfaces too; are interfaces objects, or would it be more precise to call this something like "bound_type_id"?

Reply:
This node represents `x.F` where the type of `x` is a class (or probably, eventually, a facet), not `Class.F` or `Interface.F`. So the value we're storing here is always the object argument -- the thing that will become `self` in the call, or `*self` for an `addr self` parameter.

Added some comments to explain.

Reply:
FWIW, I think this is still what I meant, but essentially yes: an instance given an interface (class, facet, etc) would still be called an object.